src/BoundaryConditions/update_boundary_conditions.jl

```diff
@@ -1,13 +1,13 @@
-update_boundary_condition!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
+update_boundary_conditions!(bc::Union{BoundaryCondition, Nothing}, args...) = nothing
 
 function update_boundary_conditions!(bcs::FieldBoundaryConditions, field, model)
-    update_boundary_condition!(bcs.west, field, model, Val(:west))
-    update_boundary_condition!(bcs.east, field, model, Val(:east))
-    update_boundary_condition!(bcs.south, field, model, Val(:south))
-    update_boundary_condition!(bcs.north, field, model, Val(:north))
-    update_boundary_condition!(bcs.bottom, field, model, Val(:bottom))
-    update_boundary_condition!(bcs.top, field, model, Val(:top))
-    update_boundary_condition!(bcs.immersed, field, model, Val(:immersed))
+    update_boundary_conditions!(bcs.west, field, model, Val(:west))
```

Comment:
Why this change? Don't these functions update only one boundary condition?

I also forget what these functions do (what's an "update"?) Is this for field time serieses?

Reply:
This is a new function from this PR intended to allow the matching schemes to be "updated", for example to let them calculate the mean wall normal flow.

Originally I had a mix for `update_boundary_condition!` and  `update_boundary_conditions!` which was confusing, and it seems this way is consistent with the rest of the code, for example `fill_halo_regions!` works on both one field or many.

Reply:
Okay yeah that makes sense, the only other reason I can think to keep it this way is that the tuple/namedtuple version below works for both tuples/namedtuples of both fields and boundary conditions in this form, where as if we change it then that won't work.

Reply:
I have fixed this now by remaining all to `update_boundary_condition!